tests/v3/test_store/test_remote.py

```diff
@@ -70,5 +71,42 @@ async def test_basic(s3):
     await store.set("foo", Buffer.from_bytes(data))
     assert await store.exists("foo")
     assert (await store.get("foo")).to_bytes() == data
-    out = await store.get_partial_values([("foo", (1, None))])
+    out = await store.get_partial_values(
+        prototype=default_buffer_prototype, key_ranges=[("foo", (1, None))]
+    )
     assert out[0].to_bytes() == data[1:]
+
+
+class TestRemoteStoreS3(StoreTests[RemoteStore]):
+    store_cls = RemoteStore
+
+    @pytest.fixture(scope="function")
+    def store_kwargs(self) -> dict[str, str | bool]:
+        return {
+            "mode": "w",
+            "endpoint_url": endpoint_uri,
+            "anon": False,
+            "url": f"s3://{test_bucket_name}",
+        }
+
+    @pytest.fixture(scope="function")
+    def store(self, store_kwargs: dict[str, str | bool]) -> RemoteStore:
+        return self.store_cls(**store_kwargs)
+
+    def get(self, store: RemoteStore, key: str) -> Buffer:
+        return Buffer.from_bytes(store._fs.get_mapper()[os.path.join(store.path, key)])
```

Comment:
Why use `mapper()` here and two line below - just to give an alternative way to do what's encoded in RemoteStore?

`os.path.join` will be wrong on windows, "/" would be better: `f"{store.path}/{key}"` (assuming `path` doesn't end with a "/").

Reply:
> Why use mapper() here and two line below - just to give an alternative way to do what's encoded in RemoteStore?

Yes, the logic is that if there are separate errors in `store.get` and `store.set`, then testing both methods in the same test case will mask one of the two, so the `StoreTests` class defines `set` and `get` that should not use the eponymous store methods

Reply:
the mapper is replaced by simpler approaches, and I'm not using `os.path.join` any more